services/web/server/tests/unit/with_dbs/09/test_projects_01.py

```diff
@@ -852,28 +852,33 @@ async def test_delete_project(
     user_project,
     expected,
     storage_subsystem_mock,
-    mocked_director_subsystem,
+    mocked_director_api,
     catalog_subsystem_mock,
     fake_services,
 ):
     # DELETE /v0/projects/{project_id}
 
     fakes = fake_services(5)
-    mocked_director_subsystem[
-        "get_running_interactive_services"
-    ].return_value = future_with_result(fakes)
+    mocked_director_api["get_running_interactive_services"].return_value = fakes
 
     await _delete_project(client, user_project, expected)
     await asyncio.sleep(2)  # let some time fly for the background tasks to run
 
     if expected == web.HTTPNoContent:
-        mocked_director_subsystem[
-            "get_running_interactive_services"
-        ].assert_called_once()
-        calls = [
-            call(client.server.app, service["service_uuid"], True) for service in fakes
+        mocked_director_api["get_running_interactive_services"].assert_called_once()
+
+        expected_calls = [
+            call(
+                # app=
+                client.server.app,
+                # service_uuid=
+                service["service_uuid"],
+                # save_state=
+                True,
```

Comment:
I think you can put the keys in here as well. Have you tried that?

Reply:
yes, i tried because i thought it was more informative but the tests checks exactly how it was called ... and the call used positional arguments, not key-arguments
IMO it would be more consistent if mock library would be so strict only with required keyword/positional arguments ... but that is the way it is ... so I commented them for clarity